qiita_db/software.py

```diff
@@ -515,18 +578,15 @@ def values(self):
 
         Returns
         -------
-        dict
+        dict of {str: object}
             Dictionary with the parameters values keyed by parameter name
         """
         with qdb.sql_connection.TRN:
-            sql = "SELECT * FROM qiita.{0} WHERE parameters_id = %s".format(
-                self._table)
+            sql = """SELECT parameter_set
+                     FROM qiita.default_parameter_set
+                     WHERE default_parameter_set_id = %s"""
             qdb.sql_connection.TRN.add(sql, [self.id])
-            # There should be only one row
-            result = dict(qdb.sql_connection.TRN.execute_fetchindex()[0])
-            del result["parameters_id"]
-            del result["param_set_name"]
-            return result
+            return qdb.sql_connection.TRN.execute_fetchlast()
```

Comment:
Nope, it is only retrieving a single value from the database, `parameter_set` which is of JSON type and psycopg2 is so awesome that transforms it to a dict directly :boom: Also there is already a test for it.